python/ray/autoscaler/autoscaler.py

```diff
@@ -426,6 +426,8 @@ def _get_node_type_specific_fields(self, node_id: str,
         return fields
 
     def _get_node_specific_docker_config(self, node_id):
+        if "docker" not in self.config:
+            return {}
         docker_config = copy.deepcopy(self.config.get("docker", {}))
         node_specific_docker = self._get_node_type_specific_fields(
```

Comment:
What are the semantics supposed to be if there is no global docker config, but every node has a node specific config?

Reply:
nvm this isn't possible because the node specific config isn't expressive enough to avoid using a global docker config